webapp/main.py

```diff
@@ -411,6 +413,17 @@ def poll():
                 raise RuntimeError("Need to login")
         except:
             raise RuntimeError("Need to login")
+
+@app.route('/toggle-server')
+def toggleServer():
+    global current_service_address
+    if current_service_address == compute_service_address:
+        current_service_address = dev_service_address
+        return "development"
+    else:
+        current_service_address = compute_service_address
+        return "production"
+
```

Comment:
Can you store the `current_service_address` variable inside `session`? Right now the `current_service_address` is global, meaning that if one user uses the button, it'll change the service address for everyone. `session` is a cookie variable, meaning it's tied to individual users/browsers. Check out lines 95-97 for usage

Reply:
ok, this should be fixed now